src/dbnode/storage/stats/query_stats.go

```diff
@@ -45,7 +46,8 @@ var (
 
 // QueryStats provides an interface for updating query stats.
 type QueryStats interface {
```

Comment:
Can we rename this to `QueryLimits`? QueryStats is a little non-descriptive

Reply:
The limiting is optional though, and the default implementation is the metrics